oak-auth-external/src/main/java/org/apache/jackrabbit/oak/spi/security/authentication/external/impl/principal/ExternalPrincipalConfiguration.java

```diff
@@ -75,34 +74,59 @@
  * @see <a href="https://issues.apache.org/jira/browse/OAK-4101">OAK-4101</a>
  */
 @Component(
-        metatype = true,
-        label = "Apache Jackrabbit Oak External PrincipalConfiguration",
-        immediate = true
+        immediate = true,
+        service = {
+                PrincipalConfiguration.class,
+                SecurityConfiguration.class
+        },
+        property = {
```

Comment:
same as above.

Reply:
In this case we don't have `ConfigurationPolicy.REQUIRED`. That means the component is started without configuration. Without having these default service properties, they would be missing from a service without configuration. AFAIK the property `oak.security.name` is used by consumers of this service to verify the presence of various implementations. Thus, in the absence of this property and the service running without explicit configuration, there is a high chance for a regression. I recommend keeping these properties.